system/datatypes/peer_reviewed_article.php

```diff
@@ -434,6 +482,13 @@ public function editAction(HyphaRequest $request) {
 		$this->xml->find(self::FIELD_NAME_EXCERPT)->setHtml($form->dataFor(self::FIELD_NAME_EXCERPT));
 		$this->xml->find(self::FIELD_NAME_METHOD)->setHtml($form->dataFor(self::FIELD_NAME_METHOD));
 		$this->xml->find(self::FIELD_NAME_SOURCES)->setHtml($form->dataFor(self::FIELD_NAME_SOURCES));
+		if ($newImage) {
+			$this->xml->documentElement->getOrCreate(self::FIELD_NAME_FEATURED_IMAGE)->setText($newImage);
+		}
+
+		if ($orgImg !== '' && $form->dataFor(self::FIELD_NAME_FEATURED_IMAGE_DELETED)) {
+			unlink($orgImg);
```

Comment:
Why wait until here to unlink? If you unlink in the if above, you probably do not need the `$orgImg` variable at all (since before setting a new image you can look at the original value in the XML).

Also, this still unlinks only the resized version, not the original (if present).